src/source/from-shift.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+
+function makeEager(obj) {
```

Comment:
Documentation would be really useful, here and in other functions.

Reply:
And yes, I realize that the Rust code you're porting didn't have any. Please feel free to blame the authors and/or reviewers of such code, starting with me :)

Reply:
I blame the authors and/or reviewers, yes :) I'm not sure I'm the best person to describe them as I mostly just transliterated into JavaScript.

Reply:
Fair enough. Can you just file an issue so that we do it eventually?

Reply:
Sure. (just to clarify, I'm not doing it out of principle, I just really don't understand almost half of transformations / helpers here :) )